src/cljs/athens/db.cljs

```diff
@@ -249,6 +249,24 @@
           (recur (get ch (dec n))))))))
 
 
+(defn get-id
+  [uid]
+  (-> (d/q '[:find ?id
+             :in $ ?uid
+             :where [?id :block/uid ?uid]]
+           @dsdb
+           uid)
+      ffirst))
```

Comment:
Should be `(d/q '[:find ?id .`

Mind the period.

https://docs.datomic.com/on-prem/query.html#find-specifications

Reply:
I would actually just add this below `db/e-by-av`. Direct indexing is faster.

```
(defn v-by-ea
  [e a]
  (-> (d/datoms @dsdb :eavt e a) first :v))
```

Reply:
We don't need the `get-id` function at all now since all it does is get an id given a `uid`. Which the `e-by-av` function does perfectly.

Reply:
It doesn't look like we need `v-by-ea` for this, and we can use `e-by-av`. Should we add it for future use?

Reply:
Oh good point @itsrainingmani. Yeah, add `v-by-ea` and add it to `.carve_ignore`